src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -610,23 +611,53 @@ protected void _addFields(Map<String, POJOPropertyBuilder> props)
     protected void _addCreators(Map<String, POJOPropertyBuilder> props)
     {
         // can be null if annotation processing is disabled...
-        if (!_useAnnotations) {
-            return;
-        }
-        for (AnnotatedConstructor ctor : _classDef.getConstructors()) {
-            if (_creatorProperties == null) {
-                _creatorProperties = new LinkedList<POJOPropertyBuilder>();
+        if (_useAnnotations) {
+            for (AnnotatedConstructor ctor : _classDef.getConstructors()) {
+                if (_creatorProperties == null) {
+                    _creatorProperties = new LinkedList<POJOPropertyBuilder>();
+                }
+                for (int i = 0, len = ctor.getParameterCount(); i < len; ++i) {
+                    _addCreatorParam(props, ctor.getParameter(i));
+                }
             }
-            for (int i = 0, len = ctor.getParameterCount(); i < len; ++i) {
-                _addCreatorParam(props, ctor.getParameter(i));
+            for (AnnotatedMethod factory : _classDef.getFactoryMethods()) {
+                if (_creatorProperties == null) {
+                    _creatorProperties = new LinkedList<POJOPropertyBuilder>();
+                }
+                for (int i = 0, len = factory.getParameterCount(); i < len; ++i) {
+                    _addCreatorParam(props, factory.getParameter(i));
+                }
             }
         }
-        for (AnnotatedMethod factory : _classDef.getFactoryMethods()) {
-            if (_creatorProperties == null) {
-                _creatorProperties = new LinkedList<POJOPropertyBuilder>();
-            }
-            for (int i = 0, len = factory.getParameterCount(); i < len; ++i) {
-                _addCreatorParam(props, factory.getParameter(i));
+        if (_classDef.getType().isRecordType()) {
+            List<String> recordComponentNames = new ArrayList<String>();
+            AnnotatedConstructor canonicalCtor = JDK14Util.findRecordConstructor(
+                    _classDef, _annotationIntrospector, _config, recordComponentNames);
+
+            if (canonicalCtor != null) {
```

Comment:
Shouldn't `null` result be anomaly and result in exception? Or is the assumption that it's only returned if there are no explicitly annotated alternatives?

Reply:
Oh actually no; method does avoid returning anything if there is something that should override the canonical constructor. Naming is bit unfortunate.